### ISSUE TITLE
False positive when using `array_first` after a `count()` call

### ISSUE THREAD

**Issue by chriscpty** [kalessil/phpinspectionsea#2003]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP8.5              |

Current behaviour
--

When I call `array_first` after checking that the array has items using `count()`, PHP Inspections EA warns about a potential null case:

```php
$arr = [new \DateTime()];
if (count($arr) === 1) {
    echo array_first($arr)->format('Y-m-d');
//                      ^^ [EA] Null pointer exception may occur here. 
}
```

The same goes for using `empty()` to check.

Expected behaviour
--

Since we have already asserted that the array has at least one entry/is not empty, we know that `array_first` won't return null, so adding a null check would be redundant. Thus, there shouldn't be a warning about it.

**Comment by ea-inspections-team** [comment]
Thank you for reporting, @chriscpty! This inspection appears to be somewhat unreliable, so we are disabling it in e1263f8b0607a1f50598cc49c6dbfdcdd73f4baa for now.
